xarray/tests/test_plot.py

```diff
@@ -1024,6 +1026,35 @@ def test_2d_coord_names(self):
         with raises_regex(ValueError, 'requires 1D coordinates'):
             self.plotmethod(x='x2d', y='y2d')
 
+    def test_plot_rgb_image(self):
+        DataArray(
+            easy_array((10, 15, 3), start=0),
+            dims=['y', 'x', 'band'],
+        ).plot.imshow()
+        self.assertEqual(0, len(find_possible_colorbars()))
+
+    def test_plot_rgb_faceted(self):
+        DataArray(
+            easy_array((2, 2, 10, 15, 3), start=0),
+            dims=['a', 'b', 'y', 'x', 'band'],
+        ).plot.imshow(row='a', col='b')
+        self.assertEqual(0, len(find_possible_colorbars()))
+
+    def test_plot_rgba_image_transposed(self):
+        # We can handle the color axis being in any position
+        DataArray(
+            easy_array((4, 10, 15), start=0),
+            dims=['band', 'y', 'x'],
+        ).plot.imshow()
+
+    def test_warns_ambigious_dim(self):
+        arr = DataArray(easy_array((3, 3, 3)), dims=['y', 'x', 'band'])
+        with pytest.warns(UserWarning):
+            arr.plot.imshow()
+        # but doesn't warn if dimensions specified
+        arr.plot.imshow(rgb='band')
+        arr.plot.imshow(x='x', y='y')
+
```

Comment:
Please add tests for errors related to `rgb`.

Reply:
Done.